### PR TITLE
Add support for Rails 7

### PR DESCRIPTION
**Pull request by gdott9**
Hello,

I removed support for Rails < 6 and added support for Rails 7.
All tests are green on my computer and it works in my rails app.

It works with only some small corrections to the existing code.
It may need more work to clean up the code and use the public migration API mentioned in issue #191 but it may be enough to release a first version with Rails 7 support.

What do you think about this ?

### CODE REVIEW

File: gemfiles/rails_7.0.gemfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# This file was generated by Appraisal
+
+source "http://rubygems.org"
+
+gem "sqlite3", "~> 1.4"
+gem "rails", "~> 7.0"
```
Comment (ilyakatz) [inline gemfiles/rails_7.0.gemfile:6]: is the full `rails` required, in the past i tried to decrease dependency just to activerecord

**Comment by ilyakatz** [comment]
Cool, looks like it required a bunch of changes, so thanks for digging through it. Once this is ready to be deployed, we can release an RC version to get it tested in real life to see if anything unexpected happens.

**Comment by ilyakatz** [comment]
Rails 7 support was added by this PR https://github.com/ilyakatz/data-migrate/pull/204, I don't think this PR is relevant any more, is that correct?

**Comment by gdott9** [comment]
Yes, it seems correct. Sorry, I did not have enough time this week to work more on this. :-)
Thank you @daztmb for your PR. :+1:

**Pull request closed without merging**